### PR TITLE
Scaladoc interface fixes, js/css-only

### PR DESCRIPTION
**Pull request by heathermiller**
Fixes for bugs
- [SI-4587](https://issues.scala-lang.org/browse/SI-4587) - Hide all/show all buttons broken
- [SI-6170](https://issues.scala-lang.org/browse/SI-6170) - Left-navigation panel broken

Additionally adds scrolling/highlighting to linked-to members in Scaladoc

### CODE REVIEW

**Comment by heathermiller** [comment]
review by @VladUreche or @jsuereth

**Comment by VladUreche** [comment]
@jsuereth or @namin can you please have a look at the JS code? Thx.

File: src/compiler/scala/tools/nsc/doc/html/resource/lib/template.js [hunk 1]
```diff
@@ -2,6 +2,22 @@
...
+        return str.replace(/([;&,\.\+\*\~':"\!\^#$%@\[\]\(\)=>\|])/g, '\\$1');
+    }
+
+    // highlight and jump to selected member
+    if (window.location.hash) {
+      var temp = window.location.hash.replace('#', '');
+      var elem = '#'+escapeJquery(temp);
+
+      window.scrollTo(0, 0);
+      $(elem).parent().effect("highlight", {color: "#FFCC85"}, 3000);
```
Comment (namin) [inline src/compiler/scala/tools/nsc/doc/html/resource/lib/template.js:17]: Does Scaladoc have 'styles'? In that case, shouldn't the highlighting color be configurable?

Comment (heathermiller) [inline src/compiler/scala/tools/nsc/doc/html/resource/lib/template.js:17]: Sure, there's a style sheet, but the style sheets aren't selectable. There's only one "style" for Scaladoc (as nice as theming would be!). 

The short answer is that no one has ever reorganized the user-facing chunks of Scaladoc (i.e. the templates that generate HTML, etc) in hopes of making a theme-able Scaladoc.

Comment (heathermiller) [inline src/compiler/scala/tools/nsc/doc/html/resource/lib/template.js:17]: i.e. spacings and colors tend to be hardcoded all over the place in the templates and elsewhere

**Comment by namin** [comment]
LGTM

**Pull request merged**